src/material/tooltip/tooltip.ts

```diff
@@ -276,6 +282,7 @@ export class MatTooltip implements OnDestroy, AfterViewInit {
 
   ngAfterViewInit() {
     // This needs to happen after view init so the initial values for all inputs have been set.
+    this._viewInitialized = true;
```

Comment:
I think the only case where this would help is if the message changes quickly before `ngAfterViewInit` has fired, but I don't know if that's actually possible, or if it is, it's very unlikely. In theory it can happen if people were to go directly through the setter, but they'd have to do it inside a constructor for it to matter.

Reply:
Better to be safe, right?

Reply:
IMO it's overly-protective since people would have to actively go out of their way to make it differently, and even if it does, it won't actually break, it'll just bind some events before they're actually needed.

Reply:
Was there a reason that it hooked up in ngAfterViewInit before? My concern was that adding the call to the message and disabled setters would cause the events to be attached earlier than they were before without this check.

Reply:
I don't think there's a specific reason for it. My guess is that we had the lifecycle hook defined already so we reused it instead of adding another one.